components/root/root.jsx

```diff
@@ -154,6 +158,76 @@ export default class Root extends React.Component {
 
             activeElement.removeEventListener('blur', removeClass);
             activeElement.addEventListener('blur', removeClass, {once: true});
+        } else if (isKeyPressed(e, Constants.KeyCodes.F6)) {
+            const currentChannel = getCurrentChannel(store.getState()) || {};
+            const recentPostIdInChannel = getMostRecentPostIdInChannel(store.getState(), currentChannel.id);
+            const getPostsForThread = makeGetPostsForThread();
+            const selected = getSelectedPost(store.getState());
+            const rhsPosts = getPostsForThread(store.getState(), {rootId: selected.id});
+            let rhsLastPostId;
+
+            if (rhsPosts[0]) {
+                rhsLastPostId = rhsPosts[0].id;
+            }
+
+            var elements = [];
+            const centerPost = document.getElementById('post_' + recentPostIdInChannel);
+            const centerChannelFooter = document.getElementById('centerChannelFooter');
+            const rhsPost = document.getElementById('rhsPost_' + rhsLastPostId);
+            const rhsFooter = document.getElementById('rhsFooter');
+            const lhsHeader = document.getElementById('lhsHeader');
+            const lhsList = document.getElementById('lhsList');
+            const channelHeader = document.getElementById('channel-header');
+            const searchBox = document.getElementById('searchBox');
+
+            if (centerPost) {
+                elements.push(centerPost);
+            }
+            if (centerChannelFooter) {
+                elements.push(centerChannelFooter);
+            }
+            if (rhsPost) {
+                elements.push(rhsPost);
+            }
+            if (rhsFooter) {
+                elements.push(rhsFooter);
+            }
+            if (lhsHeader) {
+                elements.push(lhsHeader);
+            }
+            if (lhsList) {
+                elements.push(lhsList);
+            }
+            if (channelHeader) {
+                elements.push(channelHeader);
+            }
+            if (searchBox) {
+                elements.push(searchBox);
+            }
+
+            if (this.currentFocus === elements.length) {
+                this.currentFocus = 0;
+            }
+
+            const lastElement = elements[this.currentFocus - 1];
+
+            if (e.target !== lastElement && this.currentFocus !== 0) {
+                this.currentFocus = 0;
+            }
+
+            const activeElement = elements[this.currentFocus];
```

Comment:
Since this adds the focused class the same way as the handler for the tab key does, we could move this into a helper function like

```javascript
handleAccessibilityKeys = (e) => {
    if (isKeyPressed(e, Constants.KeyCodes.TAB)) {
        this.addFocusClass(e.target);
    } else if (isKeyPressed(e, Constants.KeyCodes.F6)) {
        ...
    }
}

addFocusClass = (element) => {
    element.classList.add('keyboard-focus');

    function removeClass() {
        element.classList.remove('keyboard-focus');
        element.removeEventListener('blur', removeClass);
    }

    element.addEventListener('blur', removeClass);
}
```